app/src/main/java/com/duckduckgo/app/referral/DurationBucketMapper.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2019 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.referral
+
+class DurationBucketMapper {
+
+    fun mapDurationToBucket(durationMs: Long): String {
+        for ((durationThreshold, bucketName) in durationBuckets) {
+            if (durationMs < durationThreshold) {
+                return bucketName
+            }
+        }
+        return DURATION_EXCEEDED_BUCKET_NAME
+    }
+
+    companion object {
+
+        private val durationBuckets = listOf(
```

Comment:
What's the purpose of the duration buckets? Why not send the ms time as a param? That would let you better tabulate, graph and manipulate the data.

Reply:
Sending raw durations raised a concern around fingerprinting optics. i.e., that it would _look_ like we were sending a unique ID. Even though that's not what's going on, we don't need exact numbers, and bucketing the durations removes the potential for anyone to misinterpret the data.

Reply:
We've not worried that timing a 3rd party api may _look bad_ (we know it's not actually bad) before. Seem like we're being overly cautious here but I may be missing context, I'll dig in on Asana.